rest-e2e-tests/products-api.js

```diff
@@ -843,6 +844,138 @@ describe('Products API', () => {
         })
     })
 
+    describe('POST /api/v1/products/:id/images', () => {
+        let createdProductId
+
+        before(async () => {
+            createdProductId = await createProductAndReturnId(genericProductBody)
+        })
+
+        it('requires authentication', async () => {
+            const fileBytes = Buffer.from([])
+            const response = await Streamr.api.v1.products
+                .uploadImage(createdProductId, fileBytes)
+                .call()
+            await assertResponseIsError(response, 401, 'NOT_AUTHENTICATED')
+        })
+
+        it('requires existing Product', async () => {
+            const fileBytes = Buffer.from([])
+            const response = await Streamr.api.v1.products
+                .uploadImage('non-existing-product-id', fileBytes)
+                .withAuthToken(AUTH_TOKEN)
+                .call()
+            await assertResponseIsError(response, 404, 'NOT_FOUND')
+        })
+
+        it('requires write permission on Product', async () => {
+            const fileBytes = Buffer.from([])
+            const response = await Streamr.api.v1.products
+                .uploadImage(createdProductId, fileBytes)
+                .withAuthToken(AUTH_TOKEN_2)
+                .call()
+            const json = await response.json()
+
+            assert.equal(response.status, 403)
+            assert.equal(json.code, 'FORBIDDEN')
+            assert.equal(json.operation, 'write')
+        })
+
+        it('requires parameter "file" to be a multipart file', async () => {
+            const fileBytes = 'I am not a file'
+            const response = await Streamr.api.v1.products
+                .uploadImage(createdProductId, fileBytes)
+                .withAuthToken(AUTH_TOKEN)
+                .call()
+            await assertResponseIsError(response, 400, 'PARAMETER_MISSING')
+        })
+
+        it('verifies file size', (done) => {
+            // TODO: Unpack file, how to do without going to File system?
```

Comment:
Perhaps http://tschaub.net/blog/2014/02/17/mocking-the-filesystem.html

Reply:
Thanks I'll have a look at this before landing.

Reply:
There are some changes to these tests in CORE-1240-thumbnail branch. Please review/merge those before to avoid conflicts.